plugin-core/src/main/java/appland/utils/UserLog.java

```diff
@@ -0,0 +1,56 @@
+package appland.utils;
+
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.application.PathManager;
+import com.intellij.openapi.diagnostic.Logger;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+
+public class UserLog implements Disposable {
+    private @Nullable FileOutputStream logFile;
+    private static final Logger LOG = Logger.getInstance(UserLog.class);
+
+    public UserLog(String name) {
+        try {
+            File logDir = new File(PathManager.getLogPath());
```

Comment:
I'd make the field `logFile` private and change the code like below to allow it to be final.
`log.mkdirs` returns a boolean if the directory was created. The output stream shouldn't be created if the parent dir does not exist.

```suggestion
        FileOutputStream logFile = null;
        try {
            File logDir = new File(PathManager.getLogPath());
            if (logDir.exists() || logDir.mkdirs()) {
                logFile = new FileOutputStream(new File(logDir, name), false);
            }
        } catch (FileNotFoundException e) {
            LOG.error("Failed to open log file", e);
        } finally {
            this.logFile = logFile;
        }
```